Applications/TransferLearning/CIFAR_Classification/jni/main_func.cpp

```diff
@@ -101,65 +101,93 @@ static int rangeRandom(int min, int max) {
   int remainder = RAND_MAX % n;
   int x;
   do {
-    x = rand_r(&seed);
+    x = rand_r(&SEED);
   } while (x >= RAND_MAX - remainder);
   return min + x % n;
 }
 
 /**
- * @brief     load data at specific position of file
- * @param[in] F  ifstream (input file)
- * @param[out] outVec
- * @param[out] outLabel
- * @param[in] id th data to get
- * @retval true/false false : end of data
+ * @brief     Load Image data to the given container
+ * @param[in] filename name of the file
+ * @param[out] image container to load the image data
  */
-bool getData(std::ifstream &F, std::vector<float> &outVec,
-             std::vector<float> &outLabel, unsigned int id) {
-  F.clear();
-  F.seekg(0, std::ios_base::end);
-  uint64_t file_length = F.tellg();
-  uint64_t position = (uint64_t)((feature_size + total_label_size) *
-                                 (uint64_t)id * sizeof(float));
-
-  if (position > file_length) {
-    return false;
+void getImage(const string filename, float *image) {
+  int width, height, channels;
+  uint8_t *in =
+    tflite::label_image::read_bmp(filename, &width, &height, &channels);
+
+  if (width * height * channels != image_size)
+    throw std::runtime_error("Dataset file image size error");
+
+  for (size_t i = 0; i < image_size; i++) {
+    image[i] = ((float)in[i]) / 255.0;
   }
-  F.seekg(position, std::ios::beg);
-  for (unsigned int i = 0; i < feature_size; i++)
-    F.read((char *)&outVec[i], sizeof(float));
-  for (unsigned int i = 0; i < total_label_size; i++)
-    F.read((char *)&outLabel[i], sizeof(float));
 
-  return true;
+  free(in);
 }
 
 /**
- * @brief      get data which size is batch for train
- * @param[out] outVec
- * @param[out] outLabel
- * @param[out] last if the data is finished
- * @param[in] user_data private data for the callback
- * @retval status for handling error
+ * @brief     Get the image with the given index and its label
+ * @param[in] data_path path containing the data
+ * @param[out] input container to hold the input
+ * @param[out] output container to hold the output
+ * @param[in] n index of the data to be loaded
+ * @param[in] is_val if this image is for training or validation
  */
-int getBatch_train(float **outVec, float **outLabel, bool *last,
-                   void *user_data) {
+void getNthImage(std::string data_path, float *input, float *output, int n,
+                 bool is_val = false) {
+  std::string path = data_path;
+  unsigned int data_size = total_train_data_size;
+  if (is_val)
+    data_size = total_val_data_size;
+
+  int label_idx = n / data_size;
+  int image_idx = n % data_size;
+
+  path += "train/" + total_label[label_idx] + "/";
```

Comment:
I think it is good to terminate the application if `data_path` is not ending with '/'. It would be hard to catch why this is not working

Reply:
I will intentionally add a `\` at the end of the data path.